docs/content/tutorials/plugins.md

```diff
@@ -0,0 +1,110 @@
+# Plugins
+
+The purpose of this guide is to give an introduction to the structure of a plugin and the steps required to create one. It builds upon our article describing the process of adding custom functionality. It can be seen as the proceeding steps for extracting your custom functionality to a reusable package for other developers to use.
+
+## What is a plugin?
+
+Plugins offer a way to extend and integrate the core functionality of Medusa.
+
+In most commerce solutions, you can extend the basic features but it often comes with the expense of having to build standalone web applications. Our architecture is built such that plugins run within the same process as the core eliminating the need for extra server capability and maintenance. As a result, the plugins can use all other services as dependencies and access the database.
+
+Note: You will notice that plugins vary in naming. The name should signal what functionality they provide.
+
+In the following sections, we will go through the basics of implementing a generic plugin. And finally, how to use it as part of your commerce setup.
+
+## Building a plugin
+
+Plugins are essentially a Node.js project of their own. They contain a file in root, `package.json`, that holds all metadata and dependencies of the project.
+
+The first step in creating a plugin is to initialize the Node.js project:
+
+```bash
+npm init
+```
+
+This command will ask you to fill out your project's metadata, which will eventually be used when publishing the package to NPM. After this command completes, you are ready to start implementing the functionality.
+
+### Implementation
+
+We've already gone through the process of building custom services, endpoints, and subscribers in another tutorial, so this will not be repeated. The process is the same for the logic within a plugin, meaning that the functionality is loaded as part of the core if the correct naming convention is followed.
+
```

Comment:
Maybe explain in a few more steps what should be done here:

- Explain that reader should copy the `/services/welcome.js`, `/api/index.js` and `/subscribers/welcome.js` files from tutorial and add them to `/src`
- Maybe provide a tree dump that shows which files make up the plugin directory.